@page "/cadastro-reserva"
@using Solucao.Models
@inject HttpClient Http

<div class="container my-5">
    <div class="row justify-content-center">
        <div class="col-md-6">
            <h3 class="text-center mb-4">Criar reserva</h3>

            @if (!string.IsNullOrEmpty(mensagem))
            {
                <div class="alert @(sucesso ? "alert-success" : "alert-danger")" role="alert">
                    @mensagem
                </div>
            }

            <EditForm Model="@reserve" OnValidSubmit="EnviarReserva">
                <DataAnnotationsValidator />
                <ValidationSummary class="text-danger" />

                <div class="mb-3">
                    <label for="quarto" class="form-label">Total de quartos:</label>
                    <InputDate id="quarto" @bind-Value="reserve.DateCheckIn" class="form-control" />
                </div>

                <div class="mb-3">
                    <label for="reserva" class="form-label">Total de reservas:</label>
                    <InputDate id="reserva" @bind-Value="reserve.DateCheckOut" class="form-control" />
                </div>


                <button type="submit" class="btn btn-primary w-100">Enviar</button>
            </EditForm>
        </div>
    </div>
</div>



@code {
    private ReserveModel reserve = new ReserveModel();
    private string mensagem = string.Empty;
    private bool sucesso = true; // controla o estilo da mensagem (sucesso ou erro)

    private async Task EnviarReserva()
    {
        try
        {
            var response = await Http.PostAsJsonAsync("api/reserves", reserve);

            if (response.IsSuccessStatusCode)
            {
                mensagem = "Usuário cadastrado com sucesso!";
                sucesso = true; // exibe a mensagem de sucesso
                reserve = new ReserveModel(); // limpa o formulário
            }
            else
            {
                mensagem = "Erro ao cadastrar usuário.";
                sucesso = false; // exibe a mensagem de erro
            }
        }
        catch (Exception ex)
        {
            mensagem = $"Erro na requisição: {ex.Message}";
            sucesso = false;
        }
    }
}